diff --git a/litellm/llms/openai/chat/gpt_transformation.py b/litellm/llms/openai/chat/gpt_transformation.py
index 5464b5bb7e..a29131e5aa 100644
--- a/litellm/llms/openai/chat/gpt_transformation.py
+++ b/litellm/llms/openai/chat/gpt_transformation.py
@@ -415,6 +415,7 @@ class OpenAIGPTConfig(BaseLLMModelInfo, BaseConfig):
         from litellm.types.llms.openai import ChatCompletionToolParam
 
         for i, message in enumerate(messages):
+            cast(dict, message).pop("model_id", None)
             messages[i] = cast(
                 AllMessageValues, filter_value_from_dict(message, "cache_control")  # type: ignore
             )
diff --git a/litellm/llms/openai/openai.py b/litellm/llms/openai/openai.py
index 194f29648c..58bc2b43df 100644
--- a/litellm/llms/openai/openai.py
+++ b/litellm/llms/openai/openai.py
@@ -210,6 +210,8 @@ class OpenAIConfig(BaseConfig):
     def _transform_messages(
         self, messages: List[AllMessageValues], model: str
     ) -> List[AllMessageValues]:
+        for message in messages:
+            cast(dict, message).pop("model_id", None)
         return messages
 
     def map_openai_params(
