diff --git a/litellm/llms/openai/chat/gpt_transformation.py b/litellm/llms/openai/chat/gpt_transformation.py
index 396ad5b..4d3c3ad 100644
--- a/litellm/llms/openai/chat/gpt_transformation.py
+++ b/litellm/llms/openai/chat/gpt_transformation.py
@@ -388,6 +388,7 @@ class OpenAIGPTConfig(BaseLLMModelInfo, BaseConfig):
         from litellm.types.llms.openai import ChatCompletionToolParam
 
         for i, message in enumerate(messages):
+            cast(dict, message).pop("model_id", None)
             messages[i] = cast(
                 AllMessageValues,
                 filter_value_from_dict(message, "cache_control"),  # type: ignore
diff --git a/litellm/llms/openai/openai.py b/litellm/llms/openai/openai.py
index 6b19114..7f52d1d 100644
--- a/litellm/llms/openai/openai.py
+++ b/litellm/llms/openai/openai.py
@@ -205,7 +205,11 @@ class OpenAIConfig(BaseConfig):
                 optional_params[param] = value
         return optional_params
 
-    def _transform_messages(self, messages: List[AllMessageValues], model: str) -> List[AllMessageValues]:
+    def _transform_messages(
+        self, messages: List[AllMessageValues], model: str
+    ) -> List[AllMessageValues]:
+        for message in messages:
+            cast(dict, message).pop("model_id", None)
         return messages
 
     def map_openai_params(
